src/netconf_subscribed_notifications.c

```diff
@@ -209,26 +199,104 @@ csn_send_notif(struct csn_receiver_info *recv_info, uint32_t nc_sub_id,
     message->options = NULL;
     message->options_len = 0;
 
+cleanup:
+    free(eventtime);
+    free(string_to_send);
+    return message;
+}
+
+int
+csn_send_notif(struct csn_receiver_info *recv_info, uint32_t nc_sub_id,
+        struct timespec timestamp, struct lyd_node **ly_ntf, int use_ntf)
+{
+    unyte_message_t *message = NULL;
+    struct np2srv_sub_ntf *sub;
+    int rc = SR_ERR_OK;
+    uint32_t r;
+
+    sub = sub_ntf_find(nc_sub_id, 0, 0, 0);
+    if (!sub) {
+        EINT;
+        rc = SR_ERR_INTERNAL;
+        goto cleanup;
+    }
+
+    message = csn_build_notification(timestamp, ly_ntf);
+    if (!message) {
+        EMEM;
+        rc = SR_ERR_NO_MEMORY;
+        goto cleanup;
+    }
+
     /* search transport */
     for (r = 0; r < recv_info->count; r++) {
         unyte_send(recv_info->receivers[r].udp.sender, message);
+        ATOMIC_INC_RELAXED(sub->sent_count);
     }
 
-    free(string_notification);
-
-    ATOMIC_INC_RELAXED(sub->sent_count);
+    free(message->buffer);
```

Comment:
I would prefer for this to be in the cleanup, with condition `if (message) ...`. Occurs at several places.